Improvements/gatsby 5 compatibility

mcueto opened the pull request:
Hi guys!

I was updating to `gatsby 5` from `gatsby 4` and a `graphql` lib dependency error ocurred(multiple versions were being installed).

## What was the problem it was generating?
The dependencies were conflicting and therefore the graphql schema for directus wasn't being generated.

![imagen](https://user-images.githubusercontent.com/16097410/208266807-3b244f5b-f8be-41fc-84d2-e2641688f486.png)

![imagen](https://user-images.githubusercontent.com/16097410/208266838-f1b5e81d-33ee-4713-8d81-dfd4e67a8cb5.png)


## How did I fix it?
I just updated the dependencies in the plugin to it's 5.x versions, that fixed the problem.

## Which problems still exists with this approach?
There may be breaking changes if you try to install this package on a gatsby 4 website, I tested setting the requirements as `^4` and  `4 - 5` and in both cases `npm` was installing the last version, so no gatsby 4 compatibility and that could be an issue.

I ended up setting them as `5.2.0` which is the last stable version of both dependencies, with that it works as expected on `gatsby 5`.

to install it use
```bash
npm install https://github.com/mcueto/gatsby-source-directus/tarball/improvements/gatsby-5-compatibility
```

and then you will be able to query your schema again

![imagen](https://user-images.githubusercontent.com/16097410/208266922-4e4fb1db-4699-48b0-b9d6-9693a7317672.png)


## Note to the maintainers
I didn't update the package version because I didn't know whats the correct version to make it not backwards compatible or how you would solve those "breaking changes" aforementioned.

Cheers!

mcueto committed:
Update gatsby required deps versions to 5.2.0

mcueto committed:
Set gatsby required deps as ^4

mcueto committed:
Revert "Set gatsby required deps as ^4"

This reverts commit 1cd8ddc23a666203479fd549da1774a83c572ab8.

mcueto committed:
Set gatsby required deps as 4 - 5

mcueto committed:
Revert "Set gatsby required deps as 4 - 5"

This reverts commit 31b2351c6f43cc05082bc9f4bc8f99d049df71c9.

mcueto committed:
Update pnpm-lock.yaml

rijkvanzanten commented:
Should we try making the gatsby dependencies a peerDependency instead of a regular dependency so we can have the peer set to 4 / 5? 🤔 that way it's up to the project to make sure the correct versions exist for their projects and we can keep our package backwards compatible

mcueto commented:
Ok I will try that and then tell you what is the outcome

mcueto committed:
set gatsby-source-* as peerDependencies on it's versions 4 - 5

mcueto committed:
Add eslint 4 - 5 ass peerDependencies

mcueto committed:
Update pnpm-lock.yaml

mcueto committed:
Set peerDependencies versions as pipes

mcueto committed:
Add some instructions to readme.md

Gatsby 4 and 5 instructions

mcueto commented:
Added as `peerDependency` and tested on gatsby 4 and 5.

I also added some instructions to the readme.md file in order to let the users know what to do when installing for both versions.

(ps: I Think the readme file must be renamed to `README.md`.

# Running on gatsby 4
![imagen](https://user-images.githubusercontent.com/16097410/208281640-cffe4785-b320-480f-8d9a-9806c6584a6c.png)

# Running on gatsby 5
![imagen](https://user-images.githubusercontent.com/16097410/208282112-7cdbb334-ec2a-467e-a888-3b89d9a60da7.png)

mcueto committed:
Rename readme file as README.md

mcueto commented:
Hi @freekrai please approve the workflow when possible, I have been testing gatsby 5 + the Slices API + directus and it works! :partying_face:

freekrai commented:
I'll be looking over this PR this week, also want to make sure the plug-in continues to work for 4 users as well so will be testing and posting comments

msalmeron commented:
Hi! @freekrai there's any issue or concern regarding this PR not being approved yet?
We've ongoing projects that will benefit from this, now as a workaround, we are installing the plugin directly from @mcueto fork
Thanks!

msalmeron commented:
Sorry to bother you here again here, but can be the npm package updated to include these changes?

paescuj commented on package.json:
```diff
@@ -11,10 +11,13 @@
+	"peerDependencies": {
+		"gatsby-source-filesystem": "4||5",
+		"gatsby-source-graphql": "4||5",
+		"eslint": "7||8"
```
Sorry for this post-review, but why `eslint` as a peer dependency?

mcueto replied:
I will check it out during the weekend and then will add another comment in here.

rijkvanzanten commented:
> Sorry to bother you here again here, but can be the npm package updated to include these changes?

Yes! Going up now 👍🏻
